host/esutils/client_os2.go

```diff
@@ -62,108 +63,118 @@ func (os2 *os2Client) PutIndexTemplate(t *testing.T, templateConfigFile, templat
-		Index: []string{indexName},
+	deleteReq := osapi.IndicesDeleteReq{
+		Indices: []string{indexName},
 	}
 	ctx, cancel := createContext()
 	defer cancel()
-	resp, err := deleteReq.Do(ctx, os2.client)
+	resp, err := os2.client.Indices.Delete(ctx, deleteReq)
 	require.NoError(t, err)
-	require.False(t, resp.IsError(), fmt.Sprintf("OS2 error: %s", resp.String()))
-	resp.Body.Close()
-
+	require.True(t, resp.Acknowledged, fmt.Sprintf("OS2 delete index unacknowledged: %s", resp.Inspect().Response.Body))
 }
 
 func (os2 *os2Client) PutMaxResultWindow(t *testing.T, indexName string, maxResultWindow int) error {
 
-	req := opensearchapi.IndicesPutSettingsRequest{
-		Body:  strings.NewReader(fmt.Sprintf(`{"max_result_window" : %d}`, maxResultWindow)),
-		Index: []string{indexName},
+	req := osapi.SettingsPutReq{
+		Body:    strings.NewReader(fmt.Sprintf(`{"index": {"max_result_window": %d}}`, maxResultWindow)),
+		Indices: []string{indexName},
 	}
 
 	ctx, cancel := createContext()
 	defer cancel()
-	res, err := req.Do(ctx, os2.client)
+	resp, err := os2.client.Indices.Settings.Put(ctx, req)
 	require.NoError(t, err)
-
-	res.Body.Close()
+	require.True(t, resp.Acknowledged, fmt.Sprintf("OS2 put index settings unacknowledged: %s", resp.Inspect().Response.Body))
 
 	return nil
 }
 
 func (os2 *os2Client) GetMaxResultWindow(t *testing.T, indexName string) (string, error) {
-
-	req := opensearchapi.IndicesGetSettingsRequest{
-		Index: []string{indexName},
+	req := &osapi.SettingsGetReq{
+		Indices: []string{indexName},
 	}
 
 	ctx, cancel := createContext()
 	defer cancel()
-	res, err := req.Do(ctx, os2.client)
+
+	res, err := os2.client.Indices.Settings.Get(ctx, req)
 	require.NoError(t, err)
-	defer res.Body.Close()
 
-	if res.IsError() {
-		return "", errors.New(res.String())
+	indexSettings, ok := res.Indices[indexName]
+	if !ok {
+		return "", fmt.Errorf("no settings for index %q", indexName)
 	}
 
-	var data map[string]map[string]interface{}
-	err = json.NewDecoder(res.Body).Decode(&data)
+	var settingsData map[string]interface{}
+	err = json.Unmarshal(indexSettings.Settings, &settingsData)
 	require.NoError(t, err)
 
-	if info, ok := data[indexName]["settings"].(map[string]interface{}); ok {
-		return info["index"].(map[string]interface{})["max_result_window"].(string), nil
+	// Navigate into settings.index.max_result_window
+	indexObj, ok := settingsData["index"].(map[string]interface{})
+	if !ok {
+		return "", fmt.Errorf("missing or invalid 'index' section")
 	}
 
-	return "", fmt.Errorf("no settings for index %q", indexName)
+	raw, ok := indexObj["max_result_window"]
+	if !ok || raw == nil {
+		return "", fmt.Errorf("max_result_window not found in index settings")
+	}
 
+	// Handle both string and numeric values
+	switch v := raw.(type) {
+	case string:
+		return v, nil
+	case float64:
+		return fmt.Sprintf("%.0f", v), nil
+	default:
+		return "", fmt.Errorf("unexpected type for max_result_window: %T", v)
+	}
```

Comment:
bleh.  that multi-value is kinda awkward.

well.  since there's some unavoidable "not just normal json unmarshaling" in here I think it's fine to do the chain of implicit types.  otherwise, if you feel like it, something like
```go
type data struct{
  Index struct {
    Window any `json:"max_result_window"`
  } `json:"index"`
}
var out data
json.Unmarshal(indexSettings.Settings, &out)
```
probably simplifies this.